### ISSUE TITLE
SSL support needs to be optional in twisted mail, again

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5783]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5783|
|Type|release blocker: regression|
|Created|2012-07-15 10:35:18Z|
|Branch|https://github.com/twisted/twisted/tree/mail-without-ssl-5783|

r34769 / [#4963](https://github.com/twisted/twisted/issues/4963) introduced a hard dependency on OpenSSL in `twisted.mail.tap`, whereas previously it was an optional dependency.  It should be optional again.

<details><summary>Searchable metadata</summary>

```
trac-id__5783 5783
type__release_blocker__regression release blocker: regression
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_mail_without_ssl_5783 branches/mail-without-ssl-5783
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1342348518000000 1342348518000000
changetime__1342353172000000 1342353172000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [34825]) Branching to 'mail-without-ssl-5783'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/mail-without-ssl-5783)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @exarkun|
|-|-|

LGTM, as soon as the rest of the tests pass. Thanks for the review softball :-).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [34833]) Merge mail-without-ssl-5783 

Author: exarkun
Reviewer: glyph
Fixes: [#5783](https://github.com/twisted/twisted/issues/5783)

Make `twistd mail`'s SSL dependency optional again.
